packages/core-api/src/app/types.ts

```diff
@@ -190,3 +190,35 @@ export type BackstageApp = {
    */
   getRoutes(): JSX.Element[];
 };
+
+export type AppContext = {
+  /**
+   * @deprecated Will be removed
+   */
+  getPlugins(): BackstagePlugin<any, any>[];
+
+  /**
+   * Get a common or custom icon for this app.
+   */
+  getSystemIcon(key: IconKey): IconComponent;
```

Comment:
For symmetry (but not this PR) this could be a `getSystemIcons(): AppSystemIcons` (so one can add more functionality to it like listing or so later, and to separate concerns)

Reply:
If we want to do that change I think it makes sense to do as part of this PR, what do you think the benefit would be of the change? I'm thinking it could be worth it just for the capability of being able to list icons, as we'll probably want to be able to do that in the future.

Realized we should prolly freeze both this and the components object though :p